src/EnergyPlus/CurveManager.cc

```diff
@@ -6062,94 +6004,32 @@ namespace CurveManager {
-                } else if (SELECT_CASE_var == QuadLinear) {
-                    GetCurveType = "QUADLINEAR";
-                } else if (SELECT_CASE_var == Quadratic) {
-                    GetCurveType = "QUADRATIC";
-                } else if (SELECT_CASE_var == Cubic) {
-                    GetCurveType = "CUBIC";
-                } else if (SELECT_CASE_var == BiQuadratic) {
-                    GetCurveType = "BIQUADRATIC";
-                } else if (SELECT_CASE_var == QuadraticLinear) {
-                    GetCurveType = "QUADRATICLINEAR";
-                } else if (SELECT_CASE_var == CubicLinear) {
-                    GetCurveType = "CUBICLINEAR";
-                } else if (SELECT_CASE_var == BiCubic) {
-                    GetCurveType = "BICUBIC";
-                } else if (SELECT_CASE_var == TriQuadratic) {
-                    GetCurveType = "TRIQUADRATIC";
-                } else if (SELECT_CASE_var == Exponent) {
-                    GetCurveType = "EXPONENT";
-                } else if (SELECT_CASE_var == Quartic) {
-                    GetCurveType = "QUARTIC";
-                } else if (SELECT_CASE_var == FanPressureRise) {
-                    GetCurveType = "FANPRESSURERISE";
-                } else if (SELECT_CASE_var == ExponentialSkewNormal) {
-                    GetCurveType = "EXPONENTIALSKEWNORMAL";
-                } else if (SELECT_CASE_var == Sigmoid) {
-                    GetCurveType = "SIGMOID";
-                } else if (SELECT_CASE_var == RectangularHyperbola1) {
-                    GetCurveType = "RECTANGULARHYPERBOLA1";
-                } else if (SELECT_CASE_var == RectangularHyperbola2) {
-                    GetCurveType = "RECTANGULARHYPERBOLA2";
-                } else if (SELECT_CASE_var == ExponentialDecay) {
-                    GetCurveType = "EXPONENTIALDECAY";
-                } else if (SELECT_CASE_var == DoubleExponentialDecay) {
-                    GetCurveType = "DOUBLEEXPONENTIALDECAY";
-                } else if (SELECT_CASE_var == ChillerPartLoadWithLift) {
-                    GetCurveType = "CHILLERPARTLOADWITHLIFT";
-                }
-            }
+        // Returns true if errors found
+        int curveDim = PerfCurve(CurveIndex).NumDims;
+        if (std::find(validDims.begin(),validDims.end(), curveDim) != validDims.end()) {
+            // Compatible
+            return false;
         } else {
-            GetCurveType = "";
+            // Not compatible
+            ShowSevereError(routineName + objectType + "=\"" + objectName + "\"");
+            ShowContinueError("...Invalid curve for " + curveFieldText + ".");
+            std::string validString = std::to_string(validDims[0]);
+            for (std::size_t i = 1; i < validDims.size(); i++) {
+                validString += " or " + std::to_string(validDims[i]);
+            }
+            std::string plural1 = curveDim > 1 ? "s" : "";
+            std::string plural2 = validDims[validDims.size()-1] > 1 ? "s" : "";
```

Comment:
Nitpicking. Better, but when `curveDim` > 1, both `plural1` and `plural2` should be "s" regardless of how many valid dimensions there are.  If other changes are needed, then this can be fixed, otherwise, not a big deal.

Reply:
No, I got that wrong. If validDims[0] is > 1 then plural2 should be "s" regardless of how many valid dims there are.

Reply:
Since validDims[validDims.size()-1] is the last number printed (and generally the valid dims are in ascending order...though not strictly enforced) then this should handle things. If validDims[0] is > 1, then validDims[validDims.size()-1] will also be > 1.

Reply:
Oh - right - this works. Much ado about a single character.  Carry on.